MariaDB/server
# MDEV-31230: Fix CONNECT_JDBC in CMake

- Fix feature summary
- Use native CMake module instead of `CMAKE_MODULE_PATH`
- Make Java and JNI package as required

Reviewer: <daniel@mariadb.org>

- [x] *The Jira issue number for this PR is: MDEV-31230*

## How can this PR be tested?
With the we get the following outputs:
* No java installed, error reported
```cmake
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find Java (missing: Java_JAVA_EXECUTABLE Java_JAR_EXECUTABLE
  Java_JAVAC_EXECUTABLE Java_JAVAH_EXECUTABLE Java_JAVADOC_EXECUTABLE)
  (Required is at least version "1.6")
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.16/Modules/FindJava.cmake:332 (find_package_handle_standard_args)
  storage/connect/CMakeLists.txt:282 (FIND_PACKAGE)
```
* Install `openjdk-11-jdk-headless`  (405MB), `Java` found, but not `JNI`:
```cmake
-- Found Java: /usr/bin/java (found suitable version "11.0.18", minimum required is "1.6") 
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find JNI (missing: JAVA_AWT_LIBRARY JAVA_AWT_INCLUDE_PATH)
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.16/Modules/FindJNI.cmake:372 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  storage/connect/CMakeLists.txt:283 (FIND_PACKAGE)
```
* Here we need `default-jdk` (used for GUI based applications) only for building the package
```cmake
-- Found Java: /usr/bin/java (found suitable version "11.0.18", minimum required is "1.6") 
-- Found JNI: /usr/lib/jvm/default-java/lib/libjawt.so  

The following features have been enabled:
 * CONNECT_LIBXML2, Support for LIBXML2 in the CONNECT storage engine
 * CONNECT_ODBC, Support for ODBC in the CONNECT storage engine
 * CONNECT_JDBC, Support for JDBC in the CONNECT storage engine
```
* Appendix:
I couldn't see the value of using  user defined `FindJava.cmake` and `FindJNI.cmake` files instead of default modules, so removed it as stated in of description of MDEV.
  * Without above files, without `Java` we get the error:
  ```cmake
  -- Could NOT find LibXml2 (missing: LIBXML2_LIBRARY LIBXML2_INCLUDE_DIR) 
  CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
    Could NOT find Java (missing: Java_JAVA_EXECUTABLE Java_JAR_EXECUTABLE
    Java_JAVAC_EXECUTABLE Java_JAVAH_EXECUTABLE Java_JAVADOC_EXECUTABLE)
    (Required is at least version "1.6") 
  Call Stack (most recent call first):
    /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
    /usr/share/cmake-3.16/Modules/FindJava.cmake:332 (find_package_handle_standard_args)
    storage/connect/CMakeLists.txt:281 (FIND_PACKAGE)
  ```
  * With above files and no `Java` (additional file is checked as expected):
  ```cmake
  CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
    Could NOT find JNI (missing: JAVA_AWT_LIBRARY JAVA_JVM_LIBRARY
    JAVA_INCLUDE_PATH JAVA_INCLUDE_PATH2 JAVA_AWT_INCLUDE_PATH)
  Call Stack (most recent call first):
    /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
    /usr/share/cmake-3.16/Modules/FindJNI.cmake:372 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    cmake/FindJNI.cmake:12 (include)
    storage/connect/CMakeLists.txt:283 (FIND_PACKAGE)
  ```
## Basing the PR against the correct MariaDB version
